### DISCUSSION TITLE
getrandom() moving to vDSO

### DISCUSSION THREAD

**Discussion by rwails** [shadow/shadow#3413] [General]
In Linux kernel version 6.11, `getrandom()` is moving to vDSO: <https://lwn.net/Articles/980447/>. See <https://lwn.net/Articles/982605/> under the security-related changes.

This will be something we have to handle when distros eventually adopt that kernel version. I *think* it might be as easy as adding another symbol name to the patched vdso list here:

https://github.com/shadow/shadow/blob/7e1766859f3fdbc345ad92b692df061436cfd650/src/lib/shim/patch_vdso.c#L267

**Comment by sporksmith** [comment]
I think that's about right.

We have this issue open: https://github.com/shadow/shadow/issues/3362

**Reply by sporksmith** [reply]
we'd also need to write the replacement function of course. `_replacement_getrandom`
